core/templates/pages/collection-editor-page/navbar/collection-editor-navbar.directive.spec.ts

```diff
@@ -0,0 +1,505 @@
+        5: 0
+      },
+      human_readable_contributors_summary: {},
+      language_code: 'en',
+      num_views: 0,
+      objective: 'Test Objective',
+      status: 'public',
+      tags: [],
+      thumbnail_bg_color: '#cd672b',
+      thumbnail_icon_url: '/subjects/Algebra.svg',
+      title: 'a title'
+    };
+
+    sampleCollectionBackendObject1 = {
+      id: 'collection_id',
+      title: 'a title',
+      objective: 'an objective',
+      language_code: 'en',
+      tags: [],
+      category: 'a category',
+      version: 1,
+      schema_version: 1,
+      nodes: [{
+        exploration_id: 'exp_id0',
+        exploration_summary: learnerExplorationSummaryBackendDict
+      },
+      {
+        exploration_id: 'exp_id1',
+        exploration_summary: learnerExplorationSummaryBackendDict
+      }],
+      playthrough_dict: {
+        next_exploration_id: 'expId',
+        completed_exploration_ids: ['expId2']
+      }
+    };
+
+    sampleCollectionBackendObject2 = {
+      id: 'collection_id',
+      title: null,
+      objective: 'an objective',
+      language_code: 'en',
+      tags: [],
+      category: 'a category',
+      version: 1,
+      schema_version: 1,
+      nodes: [{
+        exploration_id: 'exp_id0',
+        exploration_summary: learnerExplorationSummaryBackendDict
+      },
+      {
+        exploration_id: 'exp_id1',
+        exploration_summary: learnerExplorationSummaryBackendDict
+      }],
+      playthrough_dict: {
+        next_exploration_id: 'expId',
+        completed_exploration_ids: ['expId2']
+      }
+    };
+    sampleCollection = Collection.create(
+      sampleCollectionBackendObject1);
+    sampleCollectionRights = CollectionRights.createEmptyCollectionRights();
+
+    ctrl = $injector.instantiate(directive.controller, {
+      $rootScope: $scope,
+      $scope: $scope
+    });
+  }));
+
+  beforeEach(() => {
+    testSubscriptions = new Subscription();
+    testSubscriptions.add(
+      collectionEditorStateService.onCollectionInitialized.subscribe(
+        collectionInitializedSpy));
+  });
+
+  afterEach(() => {
+    testSubscriptions.unsubscribe();
+    ctrl.$onDestroy();
+  });
+
+  it('should set properties when initialized', function() {
+    spyOn(urlService, 'getCollectionIdFromEditorUrl')
+      .and.returnValue('collectionId');
+    spyOn(collectionEditorStateService, 'getCollection')
+      .and.returnValue(sampleCollection);
+    spyOn(collectionEditorStateService, 'getCollectionRights')
+      .and.returnValue(sampleCollectionRights);
+    spyOn(undoRedoService, 'onUndoRedoChangeApplied$').and.returnValue(
+      mockUndoRedoChangeEventEmitter);
+    spyOnProperty(collectionEditorStateService, 'onCollectionInitialized')
+      .and.returnValue(mockCollectionInitializedEventEmitter);
+
+    ctrl.$onInit();
+    $rootScope.$apply();
+    mockUndoRedoChangeEventEmitter.emit();
+    $rootScope.$apply();
+    mockCollectionInitializedEventEmitter.emit();
+    $rootScope.$apply();
```

Comment:
Do we expect both of these to emit during initialization?
Aren't the emit statements supposed to be tested separately, since they emit at different events?

Reply:
Moved to separate tests.